crates/rust-client/src/sync/tags.rs

```diff
@@ -15,11 +15,15 @@ use crate::{
     Client,
 };
 
+/// Tag management methods
 impl<R: FeltRng> Client<R> {
-    /// Returns the list of note tags tracked by the client.
+    /// Returns the list of note tags tracked by the client along with their source.
     ///
     /// When syncing the state with the node, these tags will be added to the sync request and
     /// note-related information will be retrieved for notes that have matching tags.
+    /// The source of the tag is used to differentiate between tags that are added by the user, tags
+    /// that are added automatically by the client to track notes, and tags that are added for
+    /// accounts that are being tracked by the client.
```

Comment:
What do you think of changing the format to something like this?
```suggestion
///  The source of the tag indicates its origin. It helps distinguish between:
///  - Tags added manually by the user
///  - Tags automatically added by the client to track notes
///  - Tags added for accounts tracked by the client
```